### PR TITLE
Fixes Shrapnel Pinning

### LINKED ISSUE #16679: [Bug]: Bullet shrapnel at sufficiently high enough force/speed can pin people to walls, rendering them permanently pinned
### Checks

- [X] I have searched the bug with a few keywords, and I confirm this bug was not yet reported.
- [X] The round I am reporting the bug from, or I am going to talk about, has already ended.

### Location

Horizon

### Describe the issue

Reproduction case in which this bug occurred:

Arrested traitor was filled with bullet shrapnel. Traitor flatlined. Traitor's explosive implant went off post-mortem. Bullet shrapnel from gibbed traitor flung at nearby cadet, pinning them into a nearby wall due to the knockback force. They could not be fixed/removed from the wall without an admin deleting their shrapnel reference that was keeping them anchored.

Theoretically you could just roller bed + anesthetic tank + field surgery this out of them in-game but nobody thought of this in the moment. Also, it's stupid.

### How to reproduce

_No response_

### Round ID

coJ-abrk

### Staff ckey

melariara

### LINKED ISSUE #16680: [Bug]: Shrapnel sticks you to wall permanently
### Checks

- [X] I have searched the bug with a few keywords, and I confirm this bug was not yet reported.
- [X] The round I am reporting the bug from, or I am going to talk about, has already ended.

### Location

Horizon

### Describe the issue

Melariara might be able to explain this better than me, but what happened was an traitor wearing the banshee suit exploded after dying and pinned me into a wall. The wall was positioned on the Northeast of me. No matter what the first responders or surgeons did they weren't able to get the shrapnel off of me as it still pinned me to the wall. A knife wasn't able to remove the shrapnel. Then after Melariara took control of my character I was finally able to move again.

### How to reproduce

1. Be near a suicide exploding banshee suit (unless the banshee suit doesnt have that because I don't know)
2. be at 1 tile next to a wall that isnt facing east, south, north, or west
3. See if you can move

### Round ID

coJ-abrk

### Staff ckey

Melariara

### PR DESCRIPTION
**Pull request by crunchypretzels**
I can't take any of the credit for this.
image

Anyway, from my testing, this...
Fixes #16679 
Fixes #16680 
(the same issue in two different forms)

I encourage someone else to test this, just to make sure. I killed about twenty BSTs with shrapnel on my local but you never know.

### CODE REVIEW

**Comment by crunchypretzels** [comment]
!bugfix

**Comment by crunchypretzels** [comment]
!review

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 1]
```diff
@@ -418,14 +418,15 @@ emp_act
...
-					visible_message("<span class='warning'>[src] is pinned to the wall by [O]!</span>","<span class='warning'>You are pinned to the wall by [O]!</span>")
-					src.anchored = 1
-					src.pinned += O
+			if(O != ITEMSIZE_TINY)
+				if(O.loc == src && O.sharp) //Projectile is embedded and suitable for pinning.
+					var/turf/T = near_wall(dir,2)
+
+					if(T)
+						src.forceMove(T)
+						visible_message("<span class='warning'>[src] is pinned to the wall by [O]!</span>","<span class='warning'>You are pinned to the wall by [O]!</span>")
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/human_defense.dm:427]:
```suggestion
						visible_message(
							SPAN_WARNING("\The [src] is pinned to the wall by \the [O]!"),
							SPAN_WARNING("You are pinned to the wall by \the [O]!")
						)
```

**Review by SleepyGemmy — CHANGES_REQUESTED** [review]

**Review by SleepyGemmy — APPROVED** [review]

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 2]
```diff
@@ -418,14 +418,18 @@ emp_act
...
+				if(O.loc == src && O.sharp) //Projectile is embedded and suitable for pinning.
+					var/turf/T = near_wall(dir,2)
+
+					if(T)
+						src.forceMove(T)
+						visible_message(
+							SPAN_WARNING("\The [src] is pinned to the wall by \the [O]!"),
+							SPAN_WARNING("You are pinned to the wall by \the [O]!")
+						)
+						src.anchored = 1
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/human_defense.dm:431]:
```suggestion
						src.anchored = TRUE
```

File: html/changelogs/crunchypretzels-shrapnerf.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+# Optional: Remove this file after generating master changelog.  Useful for PR changelogs that won't get used again.
+delete-after: True
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes:
+  - bugfix: "Fixed shrapnel pinning people to walls. "
```
Comment (SleepyGemmy) [inline html/changelogs/crunchypretzels-shrapnerf.yml:41]:
```suggestion
  - bugfix: "Fixed shrapnel pinning people to walls."
```

**Review by SleepyGemmy — APPROVED** [review]

**Pull request merged**